src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq_vmr.c

```diff
@@ -1425,6 +1425,56 @@ static int vmr_info_query_op(struct platform_device *pdev,
 	return ret;
 }
 
+static int vmr_log_dump_op(struct xocl_xgq_vmr *xgq, char *buf, size_t *cnt,int num_recs)
+{
+	struct vmr_log log = { 0 };
+	uint32_t count = 0;
+	//char *start_ptr = buf_ptr;
+
```

Comment:
This is 90% dup of the existing code. Wait me to provide you a guide of removing dup code please.

Reply:
@prapulkrishnamurthy please refer to this working sudo code.

https://github.com/xdavidz/XRT1/commit/cb9f2f3e9f75b275007dcfcb1b53a84ac1caa436

It is a bit complex to comment one by one, so sorry for sending you an entire different PR.